pkg/utils/ssh.go

```diff
@@ -18,10 +19,67 @@ import (
 // GetSSHConnectionString returns the SSH connection string for the given public IP and certificate file path.
 func GetSSHConnectionString(publicIP, certFilePath string) string {
 	if certFilePath != "" {
-		return fmt.Sprintf("ssh %s %s@%s -i %s", constants.AnsibleSSHShellParams, constants.AnsibleSSHUser, publicIP, certFilePath)
-	} else {
-		return fmt.Sprintf("ssh %s %s@%s", constants.AnsibleSSHUseAgentParams, constants.AnsibleSSHUser, publicIP)
+		certFilePath = fmt.Sprintf("-i %s", certFilePath)
 	}
+	return fmt.Sprintf("ssh %s %s@%s %s", constants.AnsibleSSHShellParams, constants.AnsibleSSHUser, publicIP, certFilePath)
+}
+
+// GetSCPCommandString returns the SCP command string for the given source and destination paths.
+func GetSCPCommandString(certFilePath string, sourceIP, sourcePath string, destIP, destPath string, recursive, withCompression bool) (string, error) {
+	scpParams := constants.AnsibleSSHShellParams + " -B -o LogLevel=Error"
+	if sourceIP == "" && destIP == "" {
+		return "", fmt.Errorf("source or destination should be remote")
+	}
+	if sourceIP == "" && destPath == "" {
```

Comment:
the error msg confuses me: should they be both paths (sourcePath, destPath)? should this be an or(||) op?

Reply:
good one. my bad. 
fixed